ekss/api/deps.py

```diff
@@ -15,9 +15,19 @@
 
 """FastAPI dependencies (used with the `Depends` feature)"""
 
-from ..config import CONFIG
+import hvac
 
+from ekss.adapters.outbound.vault import VaultAdapter
+from ekss.config import CONFIG, VaultConfig
 
-def get_config():
-    """Get runtime configuration."""
+
+def config_injector():
+    """Injectable config, overridable for tests"""
     return CONFIG
+
+
+def get_vault(config: VaultConfig) -> VaultAdapter:
```

Comment:
You can chain dependencies:
```suggestion
def get_vault(config: VaultConfig = Depends(config_injector)) -> VaultAdapter:
```